sklearn/utils/__init__.py

```diff
@@ -1182,3 +1204,39 @@ def is_abstract(c):
     # itemgetter is used to ensure the sort does not extend to the 2nd item of
     # the tuple
     return sorted(set(estimators), key=itemgetter(0))
+
+
+def _safe_tags(estimator, key=None, default=None):
+    """Safely get estimator tags for common checks.
+
+    :class:`~sklearn.BaseEstimator` provides the estimator tags machinery.
+    However, if a compatible estimator does not inherit from this base class,
+    we should default to the default tag.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        The estimator from which to get the tag.
+    key : str, default=None
+        Tag name to get. By default (`None`), all tags are returned.
+    default : list of {str, dtype} or bool, default=None
+        When `key is not None`, if the tag was not set in the estimator, the
+        default value set in `sklearn.utils._DEFAULT_TAGS` will be returned.
+        `default` allows to overwrite the default value.
+
+    Returns
+    -------
+    tags : dict
+        The estimator tags.
+    """
+    if hasattr(estimator, "_get_tags"):
```

Comment:
regarding the structure, I think it would help to explicitly have the if/elif/else instead of relying on the `return` statements in the middle. It's not immediately obvious which statements are mutually exclusive. (I'm sure it's more obvious for the writer than for the reader ;)  )